domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/FeatureFlagsLogger.kt

```diff
@@ -0,0 +1,129 @@
+  private val enableDownloadsSupport: PlatformParameterValue<Boolean>,
+  @EnableExtraTopicTabsUi
+  private val enableExtraTopicTabsUi: PlatformParameterValue<Boolean>,
+  @EnableLearnerStudyAnalytics
+  private val enableLearnerStudyAnalytics: PlatformParameterValue<Boolean>,
+  @EnableFastLanguageSwitchingInLesson
+  private val enableFastLanguageSwitchingInLesson: PlatformParameterValue<Boolean>,
+  @EnableLoggingLearnerStudyIds
+  private val enableLoggingLearnerStudyIds: PlatformParameterValue<Boolean>,
+  @EnableEditAccountsOptionsUi
+  private val enableEditAccountsOptionsUi: PlatformParameterValue<Boolean>,
+  @EnablePerformanceMetricsCollection
+  private val enablePerformanceMetricsCollection: PlatformParameterValue<Boolean>,
+  @EnableSpotlightUi
+  private val enableSpotlightUi: PlatformParameterValue<Boolean>,
+  @EnableInteractionConfigChangeStateRetention
+  private val enableInteractionConfigChangeStateRetention: PlatformParameterValue<Boolean>,
+  @EnableAppAndOsDeprecation
+  private val enableAppAndOsDeprecation: PlatformParameterValue<Boolean>
+) {
+  /**
+   * A variable containing a list of all the feature flags in the app.
+   *
+   * @return a list of key-value pairs of [String] and [PlatformParameterValue]
+   */
+  private var featureFlagItemMap: Map<String, PlatformParameterValue<Boolean>> = mapOf(
+    DOWNLOADS_SUPPORT to enableDownloadsSupport,
+    EXTRA_TOPIC_TABS_UI to enableExtraTopicTabsUi,
+    LEARNER_STUDY_ANALYTICS to enableLearnerStudyAnalytics,
+    FAST_LANGUAGE_SWITCHING_IN_LESSON to enableFastLanguageSwitchingInLesson,
+    LOGGING_LEARNER_STUDY_IDS to enableLoggingLearnerStudyIds,
+    EDIT_ACCOUNTS_OPTIONS_UI to enableEditAccountsOptionsUi,
+    PERFORMANCE_METRICS_COLLECTION to enablePerformanceMetricsCollection,
+    SPOTLIGHT_UI to enableSpotlightUi,
+    INTERACTION_CONFIG_CHANGE_STATE_RETENTION to enableInteractionConfigChangeStateRetention,
+    APP_AND_OS_DEPRECATION to enableAppAndOsDeprecation,
+  )
+
+  /**
+   * This method can be used to override the featureFlagItemMap and sets its value to the given map.
+   *
+   * @param featureFlagItemMap denotes the map of feature flag names to their corresponding
+   * [PlatformParameterValue]s
+   */
+  fun setFeatureFlagItemMap(featureFlagItemMap: Map<String, PlatformParameterValue<Boolean>>) {
+    this.featureFlagItemMap = featureFlagItemMap
+  }
+
+  /**
+   * This method logs the name, enabled status and sync status of all feature flags to Firebase.
+   *
+   * @param sessionId denotes the id of the current appInForeground session
+   */
+  fun logAllFeatureFlags(sessionId: String) {
+    val featureFlagItemList = mutableListOf<FeatureFlagItemContext>()
+    for (flag in featureFlagItemMap) {
+      featureFlagItemList.add(
+        createFeatureFlagItemContext(flag)
+      )
+    }
+    val featureFlagContext = FeatureFlagContext.newBuilder()
+      .setSessionId(sessionId)
+      .addAllFeatureFlags(featureFlagItemList)
+      .build()
```

Comment:
UUID isn't being set here?

Reply:
This is because it hasn't been implemented yet. It will be implemented as part of the [Core Metrics TDD](https://docs.google.com/document/d/11aL2xt4aFv8ZyW9uu5QbP3hZ1XqYuLWvtZu1NuKkAxs/edit#heading=h.p47sq11zcyy1).

Reply:
In that case, can you please add a TODO here that points to a tracking issue to add the UUID property? I think it's important to make sure that this does get implemented correctly once the other part that's needed is done.

Similarly, could you also please make sure that at least one of the new tests verifies that the UUID is empty string? That test should theoretically fail once the UUID is properly incorporated and prompt an update (which is good test design).

Reply:
Done

Reply:
I don't think a test was added--did I miss it?

Reply:
I had forgotten to add it. There is now a new test to validate that the UUID is an empty string.